scss/_grid.scss

```diff
@@ -19,42 +40,20 @@ Screen 1200px+
 .row--align-right                               { justify-content:flex-end; }
 .row--order-reverse                             { flex-direction:row-reverse; }
 
-/***** Grid setup 1 - 12 *****/
-.grid, .grid-1, .grid-2, .grid-3, .grid-4,
-.grid-5, .grid-6, .grid-7, .grid-8, .grid-9,
-.grid-10, .grid-11, .grid-12                    { padding:0 20px; box-sizing:border-box; }
+/***** Grid setup *****/
+%grid-styles                                    { @include padding(0 $grid-gutter); box-sizing:border-box; }
+.grid                                           { @extend %grid-styles;flex:1 1 0%; }
+
+@for $i from 1 through $grid-columns {
+  .grid-#{$i}                                   { max-width: 100% / $grid-columns *$i; @extend %grid-styles; }
+}
 
 /***** grid ordering *****/
 .grid--order-first                              { order:-1; }
 .grid--order-last                               { order:1; }
 
-/***** Sets grid widths *****/
-.grid                                           { flex:1 1 0%; }
-.grid-1                                         { flex-grow:1; max-width:8.333%; }
-.grid-2                                         { flex-grow:2; max-width:16.667%; }
-.grid-3                                         { flex-grow:3; max-width:25%; }
-.grid-4                                         { flex-grow:4; max-width:33.333%; }
-.grid-5                                         { flex-grow:5; max-width:41.667%; }
-.grid-6                                         { flex-grow:6; max-width:50%; }
-.grid-7                                         { flex-grow:7; max-width:58.333%; }
-.grid-8                                         { flex-grow:8; max-width:66.667%; }
-.grid-9                                         { flex-grow:9; max-width:75%; }
-.grid-10                                        { flex-grow:10; max-width:83.333%; }
-.grid-11                                        { flex-grow:11; max-width:91.667%; }
-.grid-12                                        { flex-grow:12; max-width:100%; }
-
 /***** Sets grid offsets *****/
-.grid--offset-1                                 { margin-left:8.333%; }
-.grid--offset-2                                 { margin-left:16.667%; }
-.grid--offset-3                                 { margin-left:25%; }
-.grid--offset-4                                 { margin-left:33.333%; }
-.grid--offset-5                                 { margin-left:41.667%; }
-.grid--offset-6                                 { margin-left:50%; }
-.grid--offset-7                                 { margin-left:58.333%; }
-.grid--offset-8                                 { margin-left:66.666%; }
-.grid--offset-9                                 { margin-left:75%; }
-.grid--offset-10                                { margin-left:83.333%; }
-.grid--offset-11                                { margin-left:91.667%; }
+@for $i from 1 through $grid-columns - 1 { .grid--offset-#{$i} { margin-left: 100% / $grid-columns *$i; } }
```

Comment:
do this one (and any others) as you did https://github.com/OwlyStuff/amazium/pull/80/files#diff-f6ca44772c5cdca97cf5be7357416fb8R47

That one is spot-on!